app/services/data_exports.rb

```diff
@@ -0,0 +1,94 @@
+require 'fileutils'
+require 'securerandom'
+
+module DataExports
+
+  class DataExporter
+    def export_transcription(transcription_id)
+      file_generator = TranscriptionFileGenerator.new(transcription_id)
+      file_list = file_generator.generate_transcription_files
+
+      azure = AzureBlobStorage.new
+      azure.put_files_multiple(file_list)
+
+      # we are done with the files, delete the temp directory and its contents
+      file_generator.delete_temp_directory
+    end
+  end
+
+  class TranscriptionFileGenerator
+    def initialize(transcription_id)
+      @transcription = Transcription.find(transcription_id)
+      @directory_path = File.expand_path("~/transcription_files_temp/t#{transcription_id}_#{SecureRandom.uuid}")
+
+      FileUtils.mkdir_p @directory_path
+    end
+
+    def generate_transcription_files
+      file_list = []
+      blob_directory = generate_blob_directory(@transcription)
+
+      # raw data file
+      file_path = write_raw_data_to_file(@directory_path)
+      blob_path = File.join(blob_directory, "raw_data_#{@transcription.id}.json")
+      file_list.append({ :blob_path => blob_path, :file => file_path})
+
+      # consensus text file
+      file_path = write_consensus_text_to_file(@directory_path)
+      blob_path = File.join(blob_directory, "consensus_text_#{@transcription.id}.txt")
+      file_list.append({ :blob_path => blob_path, :file => file_path})
```

Comment:
I feel like there's gotta be a way to make a function for the generation of individual files because there's some repetition here. The only thing is, each file generated uses a different function for the generation of the file...